File: dashlane-project/modules/common/utils/src/main/java/com/dashlane/util/model/UserPermission.java
package com.dashlane.util.model;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import androidx.annotation.StringDef;

@StringDef({UserPermission.ADMIN, UserPermission.LIMITED, UserPermission.UNDEFINED})
@Retention(RetentionPolicy.SOURCE)
public @interface UserPermission {
    String ADMIN = "admin";
    String LIMITED = "limited";
    String UNDEFINED = "undefined";
}

File: dashlane-project/modules/common/utils/src/main/java/com/dashlane/sharing/UserPermission.kt
package com.dashlane.sharing

import androidx.annotation.StringDef

@StringDef(UserPermission.ADMIN, UserPermission.LIMITED, UserPermission.UNDEFINED)
@Retention(AnnotationRetention.SOURCE)
annotation class UserPermission {
    companion object {
        const val ADMIN: String = "admin"
        const val LIMITED: String = "limited"
        const val UNDEFINED: String = "undefined"
    }
}
